src/controller/python/chip/native/__init__.py

```diff
@@ -168,22 +199,38 @@ def Set(self, methodName: str, resultType, argumentTypes: list):
         method.argtype = argumentTypes
 
 
-_nativeLibraryHandle: ctypes.CDLL = None
+@dataclass
+class _Handle:
+    dll: ctypes.CDLL = None
+    initialized: bool = False
+
+
+_nativeLibraryHandles: dict[Library, _Handle] = {}
+
+
+def _GetLibraryHandle(lib: Library, shouldInit: bool) -> ctypes.CDLL:
```

Comment:
while we are at it, chould we change the name of `shouldInit`? I think `expectAlreadyInitialized` is a better name to express that we look for something already initialized.

This also changes the logic a bit: previous code seems to have assumed just the existence of the handle to mean `initialized` where as the new code handles it as a separate bool. I assume this is intentional?

Reply:
> This also changes the logic a bit: previous code seems to have assumed just the existence of the handle to mean initialized where as the new code handles it as a separate bool. I assume this is intentional?

Yes, this is intentional, because the existence of the handle does not mean that it was initialized. In case when `pychip_CommonStackInit()` fails, the handle exists but it is not usable. It might happen that someone has passed wrong adapter ID to the `Init()` function.